apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx

```diff
@@ -1,121 +1,337 @@
-  {
-    label: (
-      <View style={{flexDirection: 'row', alignItems: 'center'}}>
-        <Icon.YoroiNightly size={24} />
+export const OpenOrders = ({orders, loading = false}: {orders: Orders; loading?: boolean}) => {
+  const [bottomSheetState, setBottomSheetState] = React.useState<{
+    isOpen: boolean
+    title: string
+    content?: React.ReactNode
+  }>({
+    isOpen: false,
+    title: '',
+    content: '',
+  })
+  const [confirmationModal, setConfirmationModal] = React.useState(false)
+  const strings = useStrings()
+  const [spendingPassword, setSpendingPassword] = React.useState('')
 
-        <Spacer width={4} />
-
-        <Text>ADA/</Text>
-
-        <Spacer width={4} />
-
-        <Icon.Assets size={24} />
-
-        <Spacer width={4} />
-
-        <Text>USDA</Text>
-      </View>
-    ),
-
-    mainInfo: [
-      {label: 'Token price', value: '3 ADA'},
-      {label: 'Token amount', value: '3 USDA'},
-    ],
-    hiddenInfo: [
-      {
-        label: 'Min ADA',
-        value: '2 ADA',
-      },
-      {
-        label: 'Min Received',
-        value: '2.99 USDA',
-      },
-      {
-        label: 'Fees',
-        value: '2 ADA',
-      },
-    ],
-    buttonAction: () => {
-      console.log('button pressed')
-    },
-    buttonText: 'CANCEL ORDER',
-  },
-  {
-    label: (
-      <View style={{flexDirection: 'row', alignItems: 'center'}}>
-        <Icon.YoroiNightly size={24} />
-
-        <Spacer width={4} />
-
-        <Text>ADA/</Text>
-
-        <Spacer width={4} />
-
-        <Icon.Assets size={24} />
-
-        <Spacer width={4} />
-
-        <Text>USDA</Text>
-      </View>
-    ),
-
-    mainInfo: [
-      {label: 'Token price', value: '3 ADA'},
-      {label: 'Token amount', value: '3 USDA'},
-    ],
-    hiddenInfo: [
-      {
-        label: 'Min ADA',
-        value: '2 ADA',
-      },
-      {
-        label: 'Min Received',
-        value: '2.99 USDA',
-      },
-      {
-        label: 'Fees',
-        value: '2 ADA',
-      },
-    ],
-    buttonAction: () => {
-      console.log('button pressed')
-    },
-    buttonText: 'CANCEL ORDER',
-  },
-]
-
-export const OpenOrders = () => {
-  // TODO
-  // const data = useOrderByStatusOpen({
-  //   onError: (err) => {
-  //     console.log(err)
-  //   },
-  // })
+  if (loading) {
+    return <OpenOrdersSkeleton />
+  }
 
   return (
     <View style={styles.container}>
       <View style={styles.flex}>
-        {mockOpenOrders.map((order, index) => (
+        {orders.map((order) => (
           <ExpandableInfoCard
-            key={`${order.label}  ${index}`}
-            label={order.label}
-            mainInfo={order.mainInfo}
-            hiddenInfo={order.hiddenInfo}
-            onPress={order.buttonAction}
-            buttonText="CANCEL ORDER"
+            key={`${order.assetFromLabel}-${order.assetToLabel}-${order.date}`}
+            label={<Label assetFromLabel={order.assetFromLabel} assetToLabel={order.assetToLabel} />}
+            hiddenInfo={[
```

Comment:
I managed to refactor `mainInfo` but `hiddenInfo` is using internal state  from `ExpandableInfoCard` to use the expandable feature and the refactor was becoming complex. 

Honestly, I don't like shared components with so much complexity since the requirements constantly change and we can reach the point where these components are unmaintainable. That is the reason I have copied components in the files related to open and close orders.

let me know if you think otherwise please

Reply:
We can refactor `ExpandableInfoCard ` in a separate ticket of course. It affects other features, like the pool lists

Reply:
well i have an idea

Reply:
done